### DISCUSSION TITLE
Some speech zone with low-volume or high distortion are not detected

### DISCUSSION THREAD

**Discussion by olevanss** [linto-ai/whisper-timestamped#45] [Q&A] [answered]
Hi there!

I tried to use whisper_timestamped for better time stamping. My test samples are the premade voice lines recorded in one .wav file separated by 2 seconds. For me it looked like it should be an easy task for a Whisper. After it failed I moved to whisper_timestamped.

I attach the results I got. The blue graph is the waveform of my audio, while red ranges are the timelines provided  by whisper_timestamped.

![two_secs_2022_08_31_01_31_15_4914_both_channelright_silence_removed wav](https://user-images.githubusercontent.com/86475004/221841406-1dddb2d0-bba9-4f58-b41c-019167a98941.png)

![two_secs_2022_08_31_03_50_32_4944_both_channelright_silence_removed wav](https://user-images.githubusercontent.com/86475004/221841477-b46d7694-f2e2-4c16-8ced-3df0b3c067ab.png)

![two_secs_2022_08_31_04_59_28_4944_both_channelright_silence_removed wav](https://user-images.githubusercontent.com/86475004/221841501-81e45d4f-2096-47e5-b435-ad6c6c14e9f9.png)

Most frequent errors I get are: 
1. Whisper_timestamped ignores phrases in timestamping
2. It puts wrong timestamps onto them.
or
3. Attached silent periods bordering the phrase to the phrase timestamps.

So, while overall performance is good, I would like to ask if I can do something to improve performance. I tried allowing refining whisper model up to 5 sec.

**Comment by Jeronymous** [comment, marked as answer]
Can you please describe more precisely what is the problem:
- what kind of phrases are problematic? Can it be things like speech disfluencies?
- Do your 3 plots corresponds to the 3 cases you describe? (the first and last ones do not exhibit obviously wrong behaviors).

If you are using the trainscribe() function in python you can :
- try `beam_size=5, best_of=5, temperature=(0.0, 0.2, 0.4, 0.6, 0.8, 1.0)` (for cases where some phrases are missing from the transcription)
- try `trust_whisper_timestamps = False` (then no need to tune `refine_whisper_precision`, it will just recompute all timestamps)
- try a larger model

**Reply by olevanss** [reply]
1. In the first example it ignores the last phrase which has not so much volume. 
In the middle you can see some sounds which are not a speech.
![image](https://user-images.githubusercontent.com/86475004/221849341-629ba73e-f243-4f90-9154-93da39127f38.png)
So my problem with that is that I want to detect low volume phrases. Other thing I would like to do is still to timestamp sounds which are not speech. Would that be possible?

Second speech sample is a little distorted since I analyze phone call recordings. Distortion is not dramatic and it doesn't stop whisper from time stamping it more or less ok in the first half.

In the third sample the speech is not detected as well as pause is counted like a speech. The problem with this whole region is that it is pronounced a little like mumbling and contains basically "yes" said many times repeatedly.

![image](https://user-images.githubusercontent.com/86475004/221850987-b29cfe1d-a5ee-4910-98a6-a99660e06eb8.png)


2. Partially yes, they are aligned with errors. But you still can observe problem №3 in all of the samples as well as others.

I haven't tried what you written except for the larger model :)

I will try to use you advices and let you know how it worked.
Thank you for your help!
UPD: my IDE says there are no usages of trust_whisper_timestamps variable, I need to update my whisper_timestamped

**Reply by Jeronymous** [reply]
Thank you @olevanss  for the clarification.

In what you describe, there are several sources of problems, some of which directly come from the core Whisper neural network models (and might need re-training to be correctly addressed).
So solving some of these issues is currently out of the scope of whisper-timestamped (which is an extension of whisper).

And there are some particular mistakes made by Whisper models for which something can be done when trying to recover accurate timestamps.
For instance, we are currently working on another option to be more accurate around speech disfluencies, which are usually (but not always) removed by Whisper.
Disfluencies are things like fillers ("hmm",..), repetitions ("I di I did it", ...) that are generally removed by Whisper neural nets.

PS: Indeed `trust_whisper_timestamps` is a recent option (version 1.10+).

**Reply by olevanss** [reply]
Thank you!
New option on not trusting whisper_timestamps or setting beam_size helped me to improve and stabilize the output reasonably well
